src/ledger/LedgerTxn.cpp

```diff
@@ -646,10 +646,130 @@ LedgerTxn::Impl::getAllOffers()
     return offers;
 }
 
+#ifdef BEST_OFFER_DEBUGGING
+bool
+LedgerTxn::bestOfferDebuggingEnabled() const
+{
+    return getImpl()->bestOfferDebuggingEnabled();
+}
+
+bool
+LedgerTxn::Impl::bestOfferDebuggingEnabled() const
+{
+    return mParent.bestOfferDebuggingEnabled();
+}
+
+std::shared_ptr<LedgerEntry const>
+LedgerTxn::getBestOfferSlow(Asset const& buying, Asset const& selling,
+                            OfferDescriptor const* worseThan,
+                            std::unordered_set<int64_t>& exclude)
+{
+    return getImpl()->getBestOfferSlow(buying, selling, worseThan, exclude);
+}
+
+std::shared_ptr<LedgerEntry const>
+LedgerTxn::Impl::getBestOfferSlow(Asset const& buying, Asset const& selling,
+                                  OfferDescriptor const* worseThan,
+                                  std::unordered_set<int64_t>& exclude)
+{
+    std::shared_ptr<InternalLedgerEntry const> selfBest;
+    for (auto const& kv : mEntry)
+    {
+        if (kv.first.type() != InternalLedgerEntryType::LEDGER_ENTRY)
+        {
+            continue;
+        }
+
+        auto const& key = kv.first.ledgerKey();
+        if (key.type() != OFFER)
+        {
+            continue;
+        }
+
+        if (!exclude.insert(key.offer().offerID).second)
+        {
+            continue;
+        }
+
+        if (kv.second)
+        {
+            auto const& le = kv.second->ledgerEntry();
+            if (!(le.data.offer().buying == buying &&
+                  le.data.offer().selling == selling))
+            {
+                continue;
+            }
+
+            if (worseThan && !isBetterOffer(*worseThan, le))
```

Comment:
Nit: Would this be easier to read as `isBetterOffer(le, *worseThan)`? Same for `LedgerTxnRoot::Impl::
getBestOfferSlow`.

Reply:
I don't think these are equivalent. If `*worseThan == le` then 
```
!isBetterOffer(*worseThan, le) == !isBetterOffer(le, *worseThan) != isBetterOffer(le, *worseThan)
```
I didn't look at all the context, so if that condition isn't possible in this scenario then the change might be acceptable.

Reply:
It's not obvious to me why that wouldn't be possible, so I'm inclined to leave it as it is unless someone else has a clear demonstration that it isn't possible.

Reply:
Ah yeah nevermind. I was wrong.